### PR TITLE
raylib: 4.0.0 -> 4.2.0

### PR DESCRIPTION
**Pull request by nagy**
###### Description of changes

This also enables `-DCUSTOMIZE_BUILD` in order for the `-DINCLUDE_EVERYTHING` to have an effect.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/raylib/default.nix [hunk 1]
```diff
@@ -24,10 +24,21 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional alsaSupport alsa-lib
     ++ lib.optional pulseSupport libpulseaudio;
 
+  patches = [
+    # fixes glfw compile error;
+    # remove with next raylib version or when glfw 3.4.0 is released.
```
Comment (onny) [inline pkgs/development/libraries/raylib/default.nix:29]:
```suggestion
    # remove with next raylib version > 4.2.0 or when glfw 3.4.0 is released.
```

Comment (nagy) [reply pkgs/development/libraries/raylib/default.nix]: updated

File: pkgs/development/libraries/raylib/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, cmake,
+{ stdenv, lib, fetchFromGitHub, cmake, fetchpatch,
   mesa, libGLU, glfw,
   libX11, libXi, libXcursor, libXrandr, libXinerama,
   alsaSupport ? stdenv.hostPlatform.isLinux, alsa-lib,
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/raylib/default.nix:4]:
```suggestion
{ stdenv, lib, fetchFromGitHub, cmake, fetchpatch
, mesa, libGLU, glfw
, libX11, libXi, libXcursor, libXrandr, libXinerama
, alsaSupport ? stdenv.hostPlatform.isLinux, alsa-lib
```

Comment (nagy) [reply pkgs/development/libraries/raylib/default.nix]: updated

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by onny** [comment]
@ofborg build raylib

**Pull request merged**